Lib/test/test_urllib2.py

```diff
@@ -173,12 +172,29 @@ def test_password_manager(self):
 
         add("c", "http://example.com/foo", "foo", "ni")
         add("c", "http://example.com/bar", "bar", "nini")
+        add("c", "http://example.com/foo/bar", "foobar", "nibar")
 
         self.assertEqual(find_user_pass("c", "http://example.com/foo"),
                          ('foo', 'ni'))
-
         self.assertEqual(find_user_pass("c", "http://example.com/bar"),
                          ('bar', 'nini'))
+        self.assertEqual(find_user_pass("c", "http://example.com/foo/"),
+                         ('foo', 'ni'))
+        self.assertEqual(find_user_pass("c", "http://example.com/foo/bar"),
+                         ('foo', 'ni'))
+        self.assertEqual(find_user_pass("c", "http://example.com/foo/baz"),
+                         ('foo', 'ni'))
+        self.assertEqual(find_user_pass("c", "http://example.com/foobar"),
+                         (None, None))
+
+        add("c", "http://example.com/baz/", "baz", "ninini")
+
+        self.assertEqual(find_user_pass("c", "http://example.com/baz"),
+                         (None, None))
```

Comment:
I am not sure about this test. Should `http://example.com/baz` be considered a sub-URI of `http://example.com/baz/` or not? In other words, should a trailing slash be ignored?

Reply:
I am not sure.  I think, this test, which doesn't consider `/baz` to be sub-URI or `/baz/` for  HTTPPasswordMgr behavior seems safer. 

If there is any reference to this in any rfc or any other software behavior that will be helpful as a validation.